spec/lib/rmt/cli/repos_custom_spec.rb

```diff
@@ -0,0 +1,359 @@
+require 'rails_helper'
+
+# rubocop:disable RSpec/MultipleExpectations
+
+describe RMT::CLI::ReposCustom do
+  subject(:command) { described_class.start(argv) }
+
+  let(:product) { create :product }
+  let(:external_url) { 'http://example.com/repos' }
+  let(:repository_service) { RepositoryService.new }
+
+  describe '#add' do
+    context 'product exists' do
+      let(:argv) { ['add', external_url, 'foo'] }
+
+      it 'adds the repository to the database' do
+        expect { described_class.start(argv) }.to output("Successfully added custom repository.\n").to_stdout.and output('').to_stderr
+        expect(Repository.find_by(name: 'foo')).not_to be_nil
+      end
+    end
+
+    context 'invalid URL' do
+      let(:argv) { %w[add http://foo.bar foo] }
+
+      it 'adds the repository to the database' do
+        expect(described_class).to receive(:exit)
+        expect { described_class.start(argv) }.to output("Invalid URL \"http://foo.bar\" provided.\n").to_stderr.and output('').to_stdout
```

Comment:
Technically that's a valid URL. (Maybe I'm hosting my custom repos somewhere like `http://localhost/`?)